File: PHP/bubble-sort.php
<?php

    function bubbleSort(&$arr){
        $n = sizeof($arr);

        for($i = 0; $i < $n; $i++) {
            for ($j = 0; $j < $n - $i - 1; $j++) {

                if ($arr[$j] > $arr[$j+1]) // Swap
                {
                    $t = $arr[$j];
                    $arr[$j] = $arr[$j+1];
                    $arr[$j+1] = $t;
                }
        }
        }
    }

    $arr = array(12, 10, -1, 2, 4, 50, 100, 54);

    echo "Original Array:\n";
    print_r($arr);

    bubbleSort($arr);
    echo "\nArray after Bubble Sort:\n";
    print_r($arr);

?>

File: Rust/bubble_sort.rs
// Bubble Sort

fn sort(arr: &mut [i32]) {
    let len = arr.len();
    for i in 0..len {
        for j in 0..(len - 1 - i) {
            if arr[j] > arr[j+1] {
                arr.swap(j, j+1);
            }
        }
    }
}


fn main () {
    let mut test_arr:[i32; 7] = [7, 4, 1, 2, 0, 9, 6];
    println!("{:?}", test_arr);
    sort(&mut test_arr);
    println!("{:?}", test_arr);
}